File: problem242.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        charArr = [0]*26
        
        for char in s:
            charArr[ord(char)-ord('a')]+=1
            
        for char in t:
            charArr[ord(char)-ord('a')]-=1
        
        for n in charArr:
            if n!=0:
                return False
        
        return True
        
        
#Time: O(N)
# Space: O(1)

File: problem242.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        int[] counts = new int[26];
        
        if(s.Length!=t.Length)
            return false;
        
        for(int i=0; i<s.Length; i++)
        {
            counts[s[i]-'a']++;
            counts[t[i]-'a']--;
        }
        
        for(int i=0; i<26; i++)
        {
            if(counts[i]!=0)
                return false;
        }
        
        return true;
        
    }
}

//Time: O(N)
//space: O(1)
